pandas/core/common.py

```diff
@@ -1894,54 +1893,48 @@ def _possibly_convert_objects(values, convert_dates=True,
     if not hasattr(values, 'dtype'):
         values = np.array([values], dtype=np.object_)
 
-    # convert dates
-    if convert_dates and values.dtype == np.object_:
-
-        # we take an aggressive stance and convert to datetime64[ns]
-        if convert_dates == 'coerce':
-            new_values = _possibly_cast_to_datetime(
-                values, 'M8[ns]', coerce=True)
-
-            # if we are all nans then leave me alone
-            if not isnull(new_values).all():
-                values = new_values
-
-        else:
-            values = lib.maybe_convert_objects(
-                values, convert_datetime=convert_dates)
+    # If not object, do not attempt conversion
+    if not is_object_dtype(values.dtype):
+        return values
 
-    # convert timedeltas
-    if convert_timedeltas and values.dtype == np.object_:
+    # If 1 flag is coerce, ensure 2 others are False
+    conversions = (convert_dates, convert_numeric, convert_timedeltas)
+    if 'coerce' in conversions:
+        coerce_count = sum([c == 'coerce' for c in conversions])
+        if coerce_count > 1:
+            raise ValueError("'coerce' can be used at most once.")
 
-        if convert_timedeltas == 'coerce':
+        # Immediate return if coerce
+        if convert_dates == 'coerce':
+            return _possibly_cast_to_datetime(values, 'M8[ns]', coerce=True)
```

Comment:
you can change this to use ``to_datetime(....coerce=True)`` I think (it was that way a long time ago, but I changed it, but now for consitency prob better